stream-chat-android-client/src/main/java/io/getstream/chat/android/client/api2/mapping/SocketErrorMapping.kt

```diff
@@ -33,8 +34,17 @@ internal fun SocketErrorResponse.ErrorResponse.toDomain(): ErrorResponse {
         message = dto.message,
         statusCode = dto.StatusCode,
         exceptionFields = dto.exception_fields,
-        moreInfo = dto.more_info
+        moreInfo = dto.more_info,
+        details = dto.details.map { it.toDomain() },
     ).apply {
         duration = dto.duration
     }
 }
+
+internal fun SocketErrorResponse.ErrorResponse.ErrorDetail.toDomain(): ErrorDetail {
```

Comment:
Could we add a typealias for this? Or do we have to have such naming and/or nesting - we have SocketErrorResponse, which already tells us it's an error response. Then within it, we call ErrorResponse again, which is a bit redundant.

after that, we have ErrorDetail, could we just embed it as a property or something instead of another nested class and type? Or is it just how the API repsonses are detailed?

Reply:
Nesting was there before. ErrorResponse is nested within SocketErrorResponse.
Probably, as you said, in respect of API response structure. So I kept it that way.